### DISCUSSION TITLE
Inheriting notification_methods

### DISCUSSION THREAD

**Discussion by ausangshukla** [excid3/noticed#430] [General]
Need some help understanding how I can inherit notification_methods in my notifiers.

1. I have a BaseNotifier, from which all my notifiers are derived
2. In each subclass of notifier I have some notification_methods
3. Now some of these methods are the same for all Notifiers, so want to move them to the base class.
4. However wrapping the common methods in notification_methods in the base class, does not let the child class notification have those methods.

Any pointers would be much appreciated.

**Comment by excid3** [comment]
Add them to `Noticed::Notification` by including a module.

**Comment by ausangshukla** [comment]
This worked, thanks. For those who run into this here is a code snippet


```ruby
module EventExtensions
  extend ActiveSupport::Concern
  included do
    belongs_to :entity, optional: true
  end
end

module NotificationExtensions
  extend ActiveSupport::Concern

  def email_delay
    entity = event.entity
    # Randomize the delay so we dont flood aws SES
    delay_seconds = entity.entity_setting.email_delay_seconds
    delay_seconds = delay_seconds.positive? ? rand(1..delay_seconds) : rand(1..300)
    Rails.env.development? ? 0 : delay_seconds
  end
end

Rails.application.config.to_prepare do
  Noticed::Event.include EventExtensions
  Noticed::Notification.include NotificationExtensions
end

```
